fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Scope.java

```diff
@@ -20,15 +23,51 @@ public class Scope {
     private final RelationFields relationFields;
     private final Map<String, CTERelation> cteQueries = Maps.newLinkedHashMap();
 
+    private List<PlaceHolderExpr> lambdaInputs = Lists.newArrayList();
+
+    private boolean isLambdaScope = false;
+
     public Scope(RelationId relationId, RelationFields relation) {
         this.relationId = relationId;
         this.relationFields = relation;
     }
 
+    public Scope(List<PlaceHolderExpr> exprs, Scope parent) {
+        this.relationId = RelationId.anonymous();
+        List<Field> fieldList = Lists.newArrayList();
+        for (int i = 0; i < exprs.size(); ++i) {
+            fieldList.add(new Field(exprs.get(i).getName(), exprs.get(i).getType(),
+                    new TableName(TableName.LAMBDA_FUNC_TABLE, TableName.LAMBDA_FUNC_TABLE), exprs.get(i), false));
+        }
+        relationFields = new RelationFields(fieldList);
+        this.parent = parent;
+        isLambdaScope = true;
+    }
+
+    public Scope getLambdaScope() {
+        return new Scope(lambdaInputs, this);
+    }
+
+    public void putLambdaArgument(PlaceHolderExpr expr) {
+        lambdaInputs.add(expr);
+    }
+
+    public void clearLambdaArguments() {
+        lambdaInputs.clear();
+    }
+
+    public List<PlaceHolderExpr> getLambdaInputs() {
+        return lambdaInputs;
+    }
```

Comment:
it's similar to realation-scope transformer, maybe you can reuse it

Reply:
not yet. `putLambdaArgument ` collects input arrays' item type to construct lambda scope when visiting lambdaFunction. I do not get the same goals when transforming realtion-scope.